src/layer/vector/Canvas.js

```diff
@@ -339,7 +339,13 @@ export var Canvas = Renderer.extend({
 
 		if (options.stroke && options.weight !== 0) {
 			if (ctx.setLineDash) {
-				ctx.setLineDash(layer.options && layer.options._dashArray || []);
+				ctx.lineDashOffset = Number(options.dashOffset || 0);
```

Comment:
Is `Number` really neaded? We don't normaly use this in Leaflet

```suggestion
				ctx.lineDashOffset = options.dashOffset || 0;
```